datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaConnectorDiagUtils.java

```diff
@@ -61,4 +61,41 @@ public static String reduceTopicPartitionStatsResponses(Map<String, String> resp
 
     return JsonUtils.toJson(result.values());
   }
+
+  /**
+   * Reduce/Merge the KafkaConsumerOffsetsResponse responses of a collection of hosts/instances into one response
+   */
+  public static String reduceConsumerOffsetsResponses(Map<String, String> responses, Logger logger) {
+    Map<String, KafkaConsumerOffsetsResponse> result = new HashMap<>();
+
+    responses.forEach((instance, json) -> {
+      List<KafkaConsumerOffsetsResponse> responseList;
+      try {
+        responseList = JsonUtils.fromJson(json, new TypeReference<List<KafkaConsumerOffsetsResponse>>() {
+        });
+      } catch (Exception e) {
+        logger.error("Invalid response {} from instance {}", json, instance);
+        return;
```

Comment:
Aren't we supposed to return an empty String? Does this compile?

Reply:
Surprisingly, trying to return any String doesn't compile here. That's probably Java related trickery. Not sure why.